script to import from covidhub

ttung opened the pull request:
### Description
This script allows for an individual project to be imported into the aspen database.

### Test plan
Ran `AWS_PROFILE=biohub aspen-cli db import-covidhub-project --rr-project-id RR074e --covidhub-db-secret cliahub/cliahub_rds_read_prod`

ttung commented on src/py/aspen/covidhub_import/implementation.py:
```diff
@@ -0,0 +1,139 @@
+                )
+                session.add(sample)
+
+            sample.original_submission = {}
+            sample.public_identifier = (
+                f"USA/{czbid.submission_base}/{czbid.collection_date.year}"
+            )
+            sample.sample_collected_by = project.originating_lab
+            sample.sample_collector_contact_address = project.originating_address
+            sample.collection_date = czbid.collection_date
+            sample.location = project.location
+            sample.division = "California"
+            sample.country = "USA"
+            sample.organism = "SARS-CoV-2"
+
+            if sample.uploaded_pathogen_genome is None:
+                sample.uploaded_pathogen_genome = UploadedPathogenGenome(
+                    sample=sample,
+                )
+                session.add(sample.uploaded_pathogen_genome)
+            sample.uploaded_pathogen_genome.num_unambiguous_sites = (
+                consensus_genome.recovered_sites
+            )
+            sample.uploaded_pathogen_genome.num_mixed = consensus_genome.ambiguous_sites
+            sample.uploaded_pathogen_genome.num_missing_alleles = (
+                consensus_genome.missing_sites
+            )
+            sample.uploaded_pathogen_genome.sequencing_depth = (
+                consensus_genome.avg_depth
+            )
+            sample.uploaded_pathogen_genome.sequence = consensus_genome.fasta
```
this is where I could really use your help, @jackkamm

jackkamm replied:
I made a couple separate comment threads for `collection_date` and `location`. Otherwise it looks good to me.

phoenixAja commented on src/py/aspen/covidhub_import/implementation.py:
```diff
@@ -0,0 +1,137 @@
+            sample.uploaded_pathogen_genome.sequencing_depth = (
+                consensus_genome.avg_depth
+            )
+            sample.uploaded_pathogen_genome.sequence = consensus_genome.fasta
```
we don't need to do `session.add(sample)` at the end here?

ttung replied:
we only need to do that for new objects that are unlinked to any other objects.

phoenixAja reviewed (approved):
very nice!

jackkamm commented on src/py/aspen/covidhub_import/implementation.py:
```diff
@@ -0,0 +1,136 @@
+            )
+            sample.sample_collected_by = project.originating_lab
+            sample.sample_collector_contact_address = project.originating_address
+            sample.collection_date = czbid.collection_date
```
`collection_date` doesn't exist for `InternalCZBID`. I see you're outputting a warning for non-DPH samples, but it isn't clear to me whether you expect internal CZBIDs here. If you really need it, there's some complicated function to grab the date for internal czbids, though note it is bugged for samples that were PCR'd twice::
https://github.com/czbiohub/covidhub/issues/1485

ttung replied:
I think for Orange County, it's all DphCZBID.  I'll resolve when we get to that.

ttung replied:
https://app.clubhouse.io/genepi/story/123614/importing-covidhub-data-does-not-work-for-non-dphczbid

jackkamm commented on src/py/aspen/covidhub_import/implementation.py:
```diff
@@ -0,0 +1,136 @@
+            sample.sample_collected_by = project.originating_lab
+            sample.sample_collector_contact_address = project.originating_address
+            sample.collection_date = czbid.collection_date
+            sample.location = project.location
```
Note, this can be null, for example the recent VRDL samples. However, I see sample.location is `[not null]` in `schema.ddl`.

ttung replied:
https://app.clubhouse.io/genepi/story/123615/how-to-manage-vrdl-samples-in-aspen

erinstargazer commented on src/py/aspen/covidhub_import/implementation.py:
```diff
@@ -0,0 +1,136 @@
+            group = Group(
+                name=project.originating_lab,
+                address=project.originating_address,
+                # FIXME: this needs to be updated.
```
Is this going to be fixed later or did you mean to update this in the PR? If this is something we are going to update/fix later it might be good to update the note.

ttung replied:
https://app.clubhouse.io/genepi/story/123616/covidhub-importer-does-not-provide-a-sane-email-address-for-dph

erinstargazer commented on src/py/aspen/covidhub_import/implementation.py:
```diff
@@ -0,0 +1,136 @@
+        external_accessions = {
+            consensus_genome.sample_fastqs.czb_id.external_accession
+            for consensus_genome in consensus_genomes
+            if isinstance(consensus_genome.sample_fastqs.czb_id, DphCZBID)
```
Maybe this is getting filtered out at a different step but it seems odd to import samples from collaborators but not include the external accession. Is the idea that we just won't specify any project ids that aren't or public DPH projects when running this script?

Commit:
script to import from covidhub

### Description
This script allows for an individual project to be imported into the aspen database.

### Test plan
Ran `AWS_PROFILE=biohub aspen-cli db import-covidhub-project --rr-project-id RR074e --covidhub-db-secret cliahub/cliahub_rds_read_prod`
